script.js

```diff
@@ -0,0 +1,162 @@
+        array[i] = array [j]
+        array [j]= temp;
+    }
+}
+// initial conditions - excercise text load, first word highlited,  variables reset :
+
+function wordlistUpload(){
+    timeLeft.innerHTML=gameTime+' sec';
+    shuffle(words)
+    const htmlText = words.map((word, i) => `<span class="example__word" id="word${i}">${word}</span>`)
+    exampleInput.innerHTML = htmlText.join(' ')
+    activeId = 0
+    activeWord = exampleInput.querySelector(`#word${activeId}`)
+    activeWord.classList.add('active')
+    activeWord.scrollIntoView(false)
+}
+
+function wordlistRefresh(){
+    pressCounter = true
+    recordInput.value=""
+    timeLeft.innerHTML=gameTime+' sec';
+    bar.style.width = '100%'
+    cpm = 0
+    wpm = 0
+    clearTimeout(timeoutRef)
+    clearInterval(counter)
+    wordlistUpload()
+}
+
+
+// gametime countdown
+function countdown(){
+    const now = Date.now()
+    const then = now + gameTime*1000
+
+    counter = setInterval(()=> {
+        const secondsLeft = Math.round((then - Date.now())/1000);
+        const progress = Math.round((secondsLeft/gameTime*100));
+         if(secondsLeft<0){
+            clearInterval(counter)
+            return
+        }
+        timeLeft.innerHTML=secondsLeft + ' sec';
+        bar.style.width = progress + '%'
+    },1000);
+
+    timeoutRef = setTimeout(resultsLoad, gameTime*1000)
+}
+
+// function to show results and score data
+function resultsLoad(){
+    results.classList.add('visible')
+    wpmValue.innerHTML = `${wpm} wpm <span class="result__cathegory-details">(word per minute)</span>`
+    cpmValue.innerHTML = `${cpm} cpm <span class="result__cathegory-details">(characters per minute)</span>`
+
+}
+
+//word and letter correctness verification
+function spellingCheck(e){
+
+    //countdown start
+    if(pressCounter === true){countdown()}
+    pressCounter = false;
+
+    // variables - array from recorded word and excercice word
+    const excerciseWord = Array.from(words[activeId])
+    const recordedWord = Array.from(recordInput.value)
+
+    // word correctness - when space is pressed and minimum one letter recorded
+    if(e.keyCode===32 && recordInput.value.trim().length > 0){
+        activeWord.innerHTML=words[activeId]
+        // - remove 'acitve' class from previous word
+        activeWord.classList.remove('active')
+
+        // - add 'correct'/ 'fail' class to previous word
+        const status = excerciseWord.map((el, i)=> el===recordedWord[i]? 0 : 1 ).reduce((acc, curr)=>acc+curr)
+
+        //word formatig
+        //1.if word recorded is different then excercise sample - add class "fail"
+        // if word recorded the same as in excercise - add class "pass" and increment score variables
+        if(status>0 || excerciseWord.length != recordedWord.length-1)
+            {activeWord.classList.add('fail')
+        }else{
+            activeWord.classList.add('pass')
+             cpm = cpm + excerciseWord.length
+             wpm = wpm + 1}
```

Comment:
overcomplicated.
you should compare two strings, instead of mapping and reducing array of letters ;)